### PR TITLE
Generate Gherkin feature files from Model behaviors

### PR DESCRIPTION
**Pull request by Coffee2Bits**
* Added plugin to generate Gherkin feature files
* Refined the flow model for more interesting Gherkin output
* Fixed a parser exception if a None type was reported as a yaml root node

Some example output from flow.yaml:
```aac gen-gherkin-behaviors model/flow/flow.yaml test```

Delegate_Processing_DataB_Request.feature
```gherkin
# ATTENTION - This generated file is safe to modify. It will not be overwritten.

Feature: TODO: Fill out this feature description.

    Scenario: Receive DataB request and return DataC response
    Given Subsystem is running
    And ServiceTwo is running
    When The Subsystem receives a DataB request
    Then Subsystem delegates the request processing to ServiceTwo which returns a DataC response
    And Subsystem returns the DataC response
```

Process_DataA_Request.feature
```gherkin
# ATTENTION - This generated file is safe to modify. It will not be overwritten.

Feature: Process a DataA request and return a DataB response

    Scenario: Receive DataA request and return DataB response
    Given ServiceOne is running
    When The ServiceOne receives a DataA request
    Then ServiceOne processes the request into a DataB response
    And ServiceOne returns the DataB response

    Scenario: Receive invalid request
    Given ServiceOne is running
    When The ServiceOne receives request that isn&#39;t a DataA request
    Then ServiceOne returns an error response code
```

Process_DataB_Request.feature
```gherkin
# ATTENTION - This generated file is safe to modify. It will not be overwritten.

Feature: Process a DataB request and return a DataC response

    Scenario: Receive DataB request and return DataC response
    Given ServiceTwo is running
    When The ServiceTwo receives a DataB request
    Then ServiceTwo processes the request into a DataC response
    And ServiceTwo returns the DataC response
```

### CODE REVIEW

File: model/flow/ServiceOne.yaml [hunk 1]
```diff
@@ -4,19 +4,28 @@ import:
...
           type: DataB
       acceptance:
-        - scenario: go
+        - scenario: Receive DataA request and return DataB response
           given:
-            - ServiceOne is running.
+            - ServiceOne is running
           when:
-            - The user sends a DataA request 
+            - The ServiceOne receives a DataA request
```
Comment (Coffee2Bits) [inline model/flow/ServiceOne.yaml:21]:
```suggestion
            - ServiceOne receives a DataA request
```

File: model/flow/ServiceOne.yaml [hunk 2]
```diff
@@ -4,19 +4,28 @@ import:
...
           then:
-            - The user receives a DataB response
\ No newline at end of file
+            - ServiceOne processes the request into a DataB response
+            - ServiceOne returns the DataB response
+        - scenario: Receive invalid request
+          given:
+            - ServiceOne is running
+          when:
+            - The ServiceOne receives request that isn't a DataA request
```
Comment (Coffee2Bits) [inline model/flow/ServiceOne.yaml:29]:
```suggestion
            - ServiceOne receives request that isn't a DataA request
```

File: model/flow/ServiceThree.yaml [hunk 3]
```diff
@@ -13,10 +13,11 @@ model:
...
           type: DataD
       acceptance:
-        - scenario: go
+        - scenario: Receive DataC request and return DataD response
           given:
-            - The ServiceThree is running
+            - ServiceThree is running
           when:
-            - The user makes a request with DataC
+            - The ServiceThree receives a DataC request
```
Comment (Coffee2Bits) [inline model/flow/ServiceThree.yaml:20]:
```suggestion
            - ServiceThree receives a DataC request
```

File: model/flow/ServiceTwo.yaml [hunk 4]
```diff
@@ -4,20 +4,21 @@ import:
...
           type: DataC
       acceptance:
-        - scenario: go
+        - scenario: Receive DataB request and return DataC response
           given:
-            - The ServiceTwo is running
+            - ServiceTwo is running
           when:
-            - The user makes a request with DataB
+            - The ServiceTwo receives a DataB request
```
Comment (Coffee2Bits) [inline model/flow/ServiceTwo.yaml:21]:
```suggestion
            - ServiceTwo receives a DataB request
```

File: model/flow/Subsystem.yaml [hunk 5]
```diff
@@ -17,10 +17,12 @@ model:
...
       acceptance:
-        - scenario: go
+        - scenario: Receive DataB request and return DataC response
           given:
-            - The Sybsystem is running.
+            - Subsystem is running
+            - ServiceTwo is running
           when:
-            - The user issues a request with DataB
+            - The Subsystem receives a DataB request
```
Comment (Coffee2Bits) [inline model/flow/Subsystem.yaml:25]:
```suggestion
            - Subsystem receives a DataB request
```

File: model/flow/System.yaml [hunk 6]
```diff
@@ -8,27 +8,31 @@ model:
...
+        - scenario: Receive DataA request and return DataD response
           given:
-            - The System is running.
+            - System is running
+            - ServiceOne is running
+            - Subsystem is running
+            - ServiceThree is running
           when:
-            - The System receives input.in
+            - The System receives a DataA request
```
Comment (Coffee2Bits) [inline model/flow/System.yaml:35]:
```suggestion
            - System receives a DataA request
```

**Review by jondavid-black — APPROVED** [review]
Nice work.

**Pull request merged**